src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -91,17 +91,21 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
   >({});
 
   const [defaultDataSourceId, setDefaultDataSourceId] = useState<string | null>();
+  const defaultDataSourceIdRef = useRef<string | null>(null);
 
   const loadDefaultDataSourceId = useCallback(async () => {
     try {
       const scope = !!workspaces.currentWorkspace$.getValue()
         ? UiSettingScope.WORKSPACE
         : UiSettingScope.GLOBAL;
+
       const id = await uiSettings.getUserProvidedWithScope<string | null>(
         DEFAULT_DATA_SOURCE_UI_SETTINGS_ID,
         scope
       );
 
+      defaultDataSourceIdRef.current = id;
```

Comment:
Let's add it in line 95 so that the ref is always sync with latest defaultDataSourceId.